src/Hive.Core/epw_reader/EpwReader.cs

```diff
@@ -0,0 +1,93 @@
+            pManager.AddTextParameter("City, Country", "City-Country", "City and Country", GH_ParamAccess.item);
+            pManager.AddNumberParameter("GHI", "GHI", "Global Horizontal Irradiance, in Wh/m2.", GH_ParamAccess.item);
+            pManager.AddNumberParameter("DNI", "DNI", "Direct Normal Irradiance, in Wh/m2.", GH_ParamAccess.item);
+            pManager.AddNumberParameter("DHI", "DHI", "Diffuse Horizontal Irradiance, in Wh/m2.", GH_ParamAccess.item);
+            pManager.AddNumberParameter("DryBulb", "DryBulb", "Dry Bulb Temperature, in °C.", GH_ParamAccess.item);
+            pManager.AddNumberParameter("DewPoint", "DewPoint", "Dew Point Temperature, in °C.", GH_ParamAccess.item);
+            pManager.AddNumberParameter("RH", "RH", "Relative Humidity, in %.", GH_ParamAccess.item);
+            pManager.AddNumberParameter("GHIMonthly", "GHIMonthly", "Global Horizontal Irradiance (monthly total), in kWh/m2.", GH_ParamAccess.item);
+            pManager.AddNumberParameter("DryBulbMonthly", "DryBulbMonthly", "Dry Bulb Temperature (monthly average), in °C.", GH_ParamAccess.item);
+            pManager.AddNumberParameter("RHMonthly", "RHMonthly", "Relative Humidity (monthly average), in %.", GH_ParamAccess.item);
+            pManager.AddGenericObjectParameter("AmbientTemperatureCarrier", "AmbTempCarrier", "Hive.IO.EnergySystems.Air containing time resolved (hourly, i.e. 8760 entries) ambient temperature in deg C.", GH_ParamAccess.item);
+            pManager.AddIntegerParameter("Timezone", "Timezone", "Timezone of the location", GH_ParamAccess.item);
+        }
+
+
+        protected override void SolveInstance(IGH_DataAccess DA)
+        {
+            string Path = new string();
+			if (!DA.GetData(0, ref Path)) return;
+            
+
+            var Latitude = new double();
+            var Longitude = new double();
+            var City, Country = new string();
+            var GHI = new double();
+            var DNI = new double();
+            var DHI = new double();
+            var DryBulb = new double();
+            var DewPoint = new double();
+            var RH = new double();
+            var GHIMonthly = new double();
+            var DryBulbMonthly = new double();
+            var RHMonthly = new double();
+            var AmbientTemperatureCarrier = new object();
+            var Timezone = new int();
+
+            // TODO
+
+            DA.SetData(0, Latitude);
+            DA.SetData(1, Longitude);
+            DA.SetData(2, City, Country);
+            DA.SetData(3, GHI);
+            DA.SetData(4, DNI);
+            DA.SetData(5, DHI);
+            DA.SetData(6, DryBulb);
+            DA.SetData(7, DewPoint);
+            DA.SetData(8, RH);
+            DA.SetData(9, GHIMonthly);
+            DA.SetData(10, DryBulbMonthly);
+            DA.SetData(11, RHMonthly);
+            DA.SetData(12, AmbientTemperatureCarrier);
+            DA.SetData(13, Timezone);
```

Comment:
so this is computationally very slow, because grasshopper turns everything into GH_Numbers. Could you only output an epw-object by default, and these separate outputs only get written if a boolean is set to true?

Reply:
this is actually the wrong cs file, i forgot to delete it... you'll see the other one in Hive.IO. but your comment still applies. So would the idea be that this distributor only functions when the bool input is true? Should I pass the EPW object to other components, such as sia380 and solar stuff instead of the gh numbers?